### PR TITLE
Add Nace Code filtering to home page

### PR DESCRIPTION
**Pull request by evelynfoy**
# Business Analysis 🟠 Code Institute 🟠 Community Project - Pull Request 🎉🎉

## Pull Request References:
- #6 

## PR contains:
[x] Add dropdown select fields to home page to allow entering of Nace Codes to filter on.
[x] Populate the dropdown select fields with nace codes from the organisations list from the database
[x] Add functionality to filter the organisations displayed on the map to those with values that match the selections made
[x] Add button to show and hide the filter area

## Breaking changes
- This makes significant changes to the home page which could cause issues with it.

## Point of Note
- I removed the bootstrap scripts from base.html as they are inserted via bootstrap.load_js anyway and this was producing a double set that was causing problems.

### On load the filter area is closed - Click Open Filter Panel to open
FilterPanelClosed

### Make your selections from the dropdown lists provided and click the Filter button to filter the organisations displayed on the map. Click <code>Clear Filter </Code> to clear all filters. 
FilterPanelOpen

### On a mobile or small screen the filters are displayed vertically

FilterPanelOpenMobile

### Closing the filter Panel on a small screen provides more viewing space once the selection is made

FilterPanelClosedMobile
'

### CODE REVIEW

File: app/static/css/style.css [hunk 1]
```diff
@@ -208,5 +208,37 @@ a.link-text:hover {
     width: 100%;
     max-width: 100%;
     max-height: 100%;
-    margin-top: 2rem;
+    margin-top: 1rem;
 }
+
+.btn-primary {
+    background-color : var(--p-blue);
+    /* margin-top: 1rem; */
```
Comment (stefdworschak) [inline app/static/css/style.css:216]: Do we need this commented out code? If not, please remove it.

File: app/static/css/style.css [hunk 2]
```diff
@@ -208,5 +208,37 @@ a.link-text:hover {
...
+}
+
+.btn-primary, .btn-outline-primary {
+    border-color : var(--p-blue);
+    /* margin-top: 1rem; */
```
Comment (stefdworschak) [inline app/static/css/style.css:221]: Same here.

File: app/static/js/scripts.js [hunk 3]
```diff
@@ -42,25 +47,105 @@ function addOrganisations(response, map) {
...
-            map.addLayer(markers);
+            organisations = JSON.parse(this.response);
+            load_nace_codes();
+            markers = L.markerClusterGroup();
+            addMarkers(organisations)
         }
         else if (this.readyState == 4)
             alert(
-                `Unfortunately the list of organisations is currently unavailable. 
+                `Unfortunately the list of organisations is currently unavailable.
```
Comment (stefdworschak) [inline app/static/js/scripts.js:57]: Please remove the unneeded whitespace at the end of the line.

File: app/templates/base.html [hunk 4]
```diff
@@ -69,18 +69,16 @@
...
 
     {% block scripts %}
     {{ bootstrap.load_js() }}
     {% endblock %}
+
+    <!-- Leaflet JavaScript - Make sure you put this AFTER Leaflet's CSS -->
+	<script src="https://unpkg.com/leaflet@1.7.1/dist/leaflet.js" integrity="sha512-XQoYMqMTK8LvdxXYG3nZ448hOEQiglfqkJs1NOQV44cWnUrBc8PkAOcXy20w0vlaXaVUearIOBhiXZ5V3ynxwA==" crossorigin=""></script>
+    <script src="{{ url_for('static', filename='js/leaflet.markercluster-src.js')}}"></script>
 </body>
 </html>
```
Comment (stefdworschak) [inline app/templates/base.html:84]: Please add a blank line EOF

**Review by stefdworschak — APPROVED** [review]
Just a few minor comments, but looks good otherwise.

File: app/static/css/style.css (hunk 1 above)
Comment (evelynfoy) [reply app/static/css/style.css:216]: Removed

File: app/static/css/style.css (hunk 2 above)
Comment (evelynfoy) [reply app/static/css/style.css:221]: Removed

File: app/templates/base.html (hunk 4 above)
Comment (evelynfoy) [reply app/templates/base.html:82]: Added

File: app/static/js/scripts.js (hunk 3 above)
Comment (evelynfoy) [reply app/static/js/scripts.js:57]: Removed

**Pull request merged**